include/cscore.h

```diff
@@ -1,12 +1,12 @@
 /*----------------------------------------------------------------------------*/
-/* Copyright (c) FIRST 2016. All Rights Reserved.                             */
+/* Copyright (c) 2016-2017 FIRST. All Rights Reserved.                        */
 /* Open Source Software - may be modified and shared by FRC teams. The code   */
 /* must be accompanied by the FIRST BSD license file in the root directory of */
 /* the project.                                                               */
 /*----------------------------------------------------------------------------*/
 
-#ifndef CSCORE_H_
-#define CSCORE_H_
+#ifndef INCLUDE_CSCORE_H_
+#define INCLUDE_CSCORE_H_
```

Comment:
The problem is none of those directories are unique.  Every project is going to have a `src/main/native/include` folder where the headers are stored.  We are much better off prefixing with project name and stripping out the path down to include (so in the wpiutil `src/main/native/include/support/basename.h` example, the header guard would be `WPIUTIL_SUPPORT_BASENAME_H_`, which is unique and logical).